library/Icingadb/Widget/ViewModeSwitcher.php

```diff
@@ -95,24 +94,62 @@ public function setViewModeParam($viewModeParam)
      */
     public function getViewMode()
     {
-        return $this->url->getParam($this->getViewModeParam(), static::DEFAULT_VIEW_MODE);
+        return $this->getPopulatedValue($this->getViewModeParam(), $this->getDefaultViewMode());
+    }
+
+    /**
+     * Set callback to protect ids with
+     *
+     * @param   callable $protector
+     *
+     * @return  $this
+     */
+    public function setIdProtector($protector)
+    {
+        $this->protector = $protector;
+
+        return $this;
+    }
+
+    private function protectId($id)
+    {
+        if (is_callable($this->protector)) {
+            return call_user_func($this->protector, $id);
+        }
+
+        return $id;
     }
 
     protected function assemble()
     {
         $viewModeParam = $this->getViewModeParam();
         $currentViewMode = $this->getViewMode();
 
-        foreach (static::$viewModes as $viewMode => $icon) {
-            $url = $this->url->with($viewModeParam, $viewMode);
+        $this->addElement(new HiddenElement('view'));
```

Comment:
You should probably use `$viewModeParam`, no?

Reply:
✅